### PR TITLE
Refactor DAO pallet to use `DispatchError` when possible.

### PR DESCRIPTION
**Pull request by vivekvpandya**
Add pallet-did in mock for DAO pallet.

Budget: $30.

### CODE REVIEW

File: pallets/dao/Cargo.toml [hunk 1]
```diff
@@ -66,16 +66,24 @@ git = 'https://github.com/paritytech/substrate.git'
...
 version = '0.4.14'
 
-# [dependencies.pallet-did]
-# default-features = false
-# git = 'https://github.com/Delmonicos/pallet-did'
-# package = 'pallet-did'
-# version = '3.0.0'
+[dependencies.pallet-did]
+default-features = false
+git = 'https://github.com/vivekvpandya/pallet-did'
```
Comment (stojanov-igor) [inline pallets/dao/Cargo.toml]: Can we move the did pallet locally?

File: pallets/dao/src/mock.rs [hunk 2]
```diff
@@ -1,7 +1,8 @@
 use crate as pallet_dao;
 use frame_support::parameter_types;
 use frame_system as system;
-use sp_core::H256;
+use once_cell::sync::Lazy;
```
Comment (stojanov-igor) [inline pallets/dao/src/mock.rs:4]: I like this addition. Having actual accounts makes the code more readable.

File: pallets/dao/src/tests.rs [hunk 3]
```diff
@@ -519,24 +520,24 @@ fn user_can_be_removed_from_organization_it_belongs_to_member_of() {
...
-		// Ensure user 4 can be added to a DAO
-		assert_ok!(Dao::add_members(Origin::signed(1), ORG_NAME1.to_vec(), 4));
-		assert_ok!(Dao::add_members(Origin::signed(1), ORG_NAME2.to_vec(), 4));
+		// Ensure EVE can be added to a DAO
+		assert_ok!(Dao::add_members(Origin::signed(*ALICE), ORG_NAME1.to_vec(), *EVE));
+		assert_ok!(Dao::add_members(Origin::signed(*ALICE), ORG_NAME2.to_vec(), *EVE));
 
-		// Ensure the user 4 is member of 2 organizations
-		assert_eq!(Dao::member_of(4).len(), 2);
+		// Ensure the EVE is member of 2 organizations
```
Comment (stojanov-igor) [inline pallets/dao/src/tests.rs]: I updated the comments to reflect the code changes.

**Review by stojanov-igor — COMMENTED** [review]
Looks good. Can we move the DID pallet locally into the node before we merge?

File: pallets/dao/src/tests.rs (hunk 3 above)
Comment (vivekvpandya) [reply pallets/dao/src/tests.rs]: Thanks!

**Comment by vivekvpandya** [comment]
In next PR I will use pallet-did in pallet-dao as per requirements of https://github.com/UniversalDot/universal-dot-node/issues/21

**Review by stojanov-igor — APPROVED** [review]
I don't think we will use all the functionality from the DID pallet, but we can keep it for now. 

This can be merged.

**Pull request merged**